src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -287,14 +282,12 @@ private String getRegionFromImage(Instance instance) {
 
     @Override
     public RunScriptOptions getDefaultRunScriptOptions() {
-        return Optional.ofNullable(generatedPrivateKey)
-                       .filter(StringUtils::isNotEmpty)
-                       .filter(StringUtils::isNotBlank)
+        return Optional.ofNullable(generatedKeyPair)
                        // currently, we cannot login as root user on ec2 instances anymore
                        // this is why we absolutely need to pass an authentication key
                        .map(privateKey -> RunScriptOptions.Builder.runAsRoot(false)
                                                                   .overrideLoginUser(getVmUserLogin())
-                                                                  .overrideLoginPrivateKey(privateKey))
+                                                                  .overrideLoginPrivateKey(generatedKeyPair.getValue()))
```

Comment:
I'm not sure (not familiar with  `SimpleImmutableEntry`) but I think `generatedKeyPair.getValue()` should be replaced by `privateKey.getValue()` (generatedKeyPair could be null) and maybe `privateKey` renamed to `keyPair`.

Reply:
Indeed it is cleaner to use the lambda variable here, although we cannot reach this code if generatedKeyPair is null.
Thanks for the rename spot